RELEASE-howto.md

```diff
@@ -0,0 +1,121 @@
+# Howto release PyWPS
+
+This document gives you, as PyWPS release master complete tutorial of how to get
+PyWPS release rolled up and deployed to target server, create packages etc.
+
+## PyWPS versioning
+
+PyWPS uses [Debian version naming system](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version). Every policy should be checked against it.
+
+PyWPS uses 3 numbers release description: MAJOR.MINOR.MAINTENANCE. Currently,
+the MAJOR version is `4`. Within MAJOR releases, we should aim, not to break
+backwards compatibility.
+
+Odd MINOR version numbers (0, 2, 4, 6, ...) are considered as stable, where as
+even numbers (1, 3, 5, 7, ...) are current development branches. MINOR releases
+should add new features.
+
+MAINTENANCE number should be used for bugfix releases only. No new features are
+added.
+
+For release candindates `MAJOR.MINOR.MAINTENANCE~rcX` format should be used.
+
+## Create new ticket in repository
+
+At https://github.com/geopython/pywps/issues/new create new ticket with the
+title
+
+```
+PyWPS MAJOR.MINOR.MAJOR~rcX release
+```
+
+At the moment, PyWPS uses `4` as MAJOR version number.
+
+Create new branch in your private repository
+
+```
+git checkout master # or checkout to other point in history, where you need to create new release
+git checkout -b TICKET_NUMBER_pywps-X.Y.Z-rcX-release
+```
+## Fix files, commit and create pull request
+
+* Fix the [VERSION.txt](https://github.com/geopython/pywps/blob/master/VERSION.txt) file.
+* Fix the [pywps/__init__.py](https://github.com/geopython/pywps/blob/master/pywps/__init__.py) file `__version__` attribute
+* Fix the [debian/changelog](https://github.com/geopython/pywps/blob/master/debian/changelog) file
+
+```
+git commit -m"Creating new release of PYWPS X.Y.Z~rcX fixes #TICKET_NUMBER" -a
+git push YOUR_REMOTE
+```
+
+Create pull request.
+
+## After pull request is accepted
+
+Create tag
+```
+git tag X.Y.Z~rcX
+git push --tags
+```
+
+Send PyWPS to http://pypi.python.org repository
+
+```
+cd /tmp
+git clone git@github.com:geopython/pywps.git pywps-4
+cd pywps-4
+git checkout X.Y.Z
+python setup.py bdist_egg upload
+```
```

Comment:
@ldesousa so something like the [git-flow approach](http://nvie.com/posts/a-successful-git-branching-model/)?  I'm okay with a different development approach which we can agree on here or on the mailing list.

From my experience, as long as we are not forced into a major release bump because we committed a major change to master everytime we want to push out a fix or feature, is or key importance.